src/firmware/app/control/BUILD

```diff
@@ -9,18 +9,40 @@ cc_library(
 
 cc_library(
     name = "trajectory_planner",
-    srcs = ["trajectory_planner.c"],
+    srcs = [
+        "trajectory_planner.c",
+        "trajectory_planner_impl.c",
+        "trajectory_planner_impl.h",
+    ],
     hdrs = ["trajectory_planner.h"],
     deps = [
+        "//firmware/shared:physics",
         "//firmware/shared/math:polynomial_1d",
         "//firmware/shared/math:polynomial_2d",
+        "//firmware/shared/math:tbots_math",
         "//firmware/shared/math:vector_2d",
     ],
 )
 
 cc_test(
     name = "trajectory_planner_test",
-    srcs = ["trajectory_planner_test.cpp"],
+    srcs = [
+        "trajectory_planner_impl.h",
```

Comment:
why is the `impl` header needed in the "normal" (non_impl) tests?

Reply:
I use some of the impl functions to test output of the public functions